plugins/woocommerce/src/Admin/PluginsHelper.php

```diff
@@ -618,4 +644,258 @@ public static function maybe_enqueue_scripts_for_notices_in_plugins() {
+	    		<p class="widefat">' . wp_kses_post( $notice['description'] ) . '</p>
+	    	</div>';
+		}
+	}
+
+	/**
+	 * Show notice about to expiring subscription on WC settings page.
+	 *
+	 * @return void
+	 */
+	public static function maybe_show_expiring_subscriptions_notice() {
+		if ( ! WC_Helper::is_site_connected() ) {
+			return;
+		}
+
+		if ( 'woocommerce_page_wc-settings' !== get_current_screen()->id ) {
+			return;
+		}
+
+		$notice = self::get_expiring_subscription_notice();
+
+		if ( isset( $notice['description'] ) ) {
+			echo '<div id="woo-subscription-expiring-notice" class="woo-subscription-expiring-notice woo-subscription-notices notice notice-error is-dismissible">
+	    		<p class="widefat">' . wp_kses_post( $notice['description'] ) . '</p>
+	    	</div>';
+		}
+	}
+
+	/**
+	 * Prepared a message based on user subscriptions data.
+	 *
+	 * @param array $all_subs all subscription data.
+	 * @param array $subs_to_show filtered subscriptions as condition.
+	 * @param int   $total total subscription count.
+	 * @param array $messages message.
+	 * @return string message to return.
+	 */
+	public static function get_subscriptions_message( array $all_subs, array $subs_to_show, int $total, array $messages ) {
+		if ( 1 < $total ) {
+			return sprintf(
+				$messages['different_subscriptions'],
+				esc_attr( $total ),
+				esc_url( self::WOO_SUBSCRIPTION_PAGE_URL ),
+				esc_attr( $total ),
+			);
+		}
+
+		$subscription = reset( $subs_to_show );
+		$product_id   = $subscription['product_id'];
+		// check if $all_subs has multiple subs for this product.
+		$has_multiple_subs_for_product = 1 < count(
+			array_filter(
+				$all_subs,
+				function ( $sub ) use ( $product_id ) {
+					return $product_id === $sub['product_id'];
+				}
+			)
+		);
+
+		$message_key = sprintf( '%s_manage', $has_multiple_subs_for_product ? 'multiple' : 'single' );
+		$price       = $subscription['product_price'] ?? '';
+		$expiry_date = date_i18n( 'F jS', $subscription['expires'] );
+
+		if ( isset( $messages[ $message_key ] ) ) {
+			return sprintf(
+				$messages[ $message_key ],
+				esc_attr( $subscription['product_name'] ),
+				esc_attr( $expiry_date ),
+				esc_url( self::WOO_SUBSCRIPTION_PAGE_URL ),
+				esc_attr( $price ),
+			);
+		}
+
+		return '';
+	}
+
+	/**
+	 * Get formatted notice information for expiring subscription.
+	 *
+	 * @param boolean $allowed_link whether the notice description should include a link.
+	 * @return array notice information.
+	 */
+	public static function get_expiring_subscription_notice( $allowed_link = true ) {
+		if ( ! WC_Helper::is_site_connected() ) {
+			return array();
+		}
+
+		if ( ! self::$can_show_expiring_subs_notice ) {
+			return array();
+		}
+
+		if ( ! self::should_show_notice( self::DISMISS_EXPIRING_SUBS_NOTICE ) ) {
+			return array();
+		}
+
+		$subscriptions          = WC_Helper::get_subscriptions();
```

Comment:
note that this gets connected user's subscriptions (does not consider the site it's connected from)